clientv3/concurrency/session.go

```diff
@@ -107,3 +114,9 @@ func WithTTL(ttl int) SessionOption {
 		}
 	}
 }
+
+func WithContext(ctx context.Context) SessionOption {
```

Comment:
add comments about when users should use this instead of just calling close? probably also mention that if user cancel session through this, the lease might not be revoked. we will rely on server to time it out.